src/Simplex/Messaging/Agent/Store/SQLite.hs

```diff
@@ -248,83 +268,213 @@ instance (MonadUnliftIO m, MonadError StoreError m) => MonadAgentStore SQLiteSto
-  createBcast SQLiteStore {dbConn} bId =
-    liftIOEither $
-      checkConstraint SEBcastDuplicate $
-        DB.execute dbConn "INSERT INTO broadcasts (broadcast_id) VALUES (?);" (Only bId)
+  createBcast :: SQLiteStore -> TVar ChaChaDRG -> BroadcastId -> m BroadcastId
+  createBcast SQLiteStore {dbConn} gVar bcastId = liftIOEither $ case bcastId of
+    "" -> createWithRandomId gVar create
+    bId -> checkConstraint SEBcastDuplicate $ create bId $> bId
+    where
+      create bId = DB.execute dbConn "INSERT INTO broadcasts (broadcast_id) VALUES (?);" (Only bId)
 
-  addBcastConn :: SQLiteStore -> BroadcastId -> ConnAlias -> m ()
-  addBcastConn SQLiteStore {dbConn} bId connAlias =
+  addBcastConn :: SQLiteStore -> BroadcastId -> ConnId -> m ()
+  addBcastConn SQLiteStore {dbConn} bId connId =
     liftIOEither . checkBroadcast dbConn bId $
-      getConn_ dbConn connAlias >>= \case
+      getConn_ dbConn connId >>= \case
         Left _ -> pure $ Left SEConnNotFound
         Right (SomeConn _ RcvConnection {}) -> pure . Left $ SEBadConnType CRcv
         Right _ ->
           checkConstraint SEConnDuplicate $
             DB.execute
               dbConn
               "INSERT INTO broadcast_connections (broadcast_id, conn_alias) VALUES (?, ?);"
-              (bId, connAlias)
+              (bId, connId)
 
-  removeBcastConn :: SQLiteStore -> BroadcastId -> ConnAlias -> m ()
-  removeBcastConn SQLiteStore {dbConn} bId connAlias =
+  removeBcastConn :: SQLiteStore -> BroadcastId -> ConnId -> m ()
+  removeBcastConn SQLiteStore {dbConn} bId connId =
     liftIOEither . checkBroadcast dbConn bId $
-      bcastConnExists_ dbConn bId connAlias >>= \case
+      bcastConnExists_ dbConn bId connId >>= \case
         False -> pure $ Left SEConnNotFound
         _ ->
           Right
             <$> DB.execute
               dbConn
               "DELETE FROM broadcast_connections WHERE broadcast_id = ? AND conn_alias = ?;"
-              (bId, connAlias)
+              (bId, connId)
 
   deleteBcast :: SQLiteStore -> BroadcastId -> m ()
   deleteBcast SQLiteStore {dbConn} bId =
     liftIOEither . checkBroadcast dbConn bId $
       Right <$> DB.execute dbConn "DELETE FROM broadcasts WHERE broadcast_id = ?;" (Only bId)
 
-  getBcast :: SQLiteStore -> BroadcastId -> m [ConnAlias]
+  getBcast :: SQLiteStore -> BroadcastId -> m [ConnId]
   getBcast SQLiteStore {dbConn} bId =
     liftIOEither . checkBroadcast dbConn bId $
       Right . map fromOnly
         <$> DB.query dbConn "SELECT conn_alias FROM broadcast_connections WHERE broadcast_id = ?;" (Only bId)
 
+  createIntro :: SQLiteStore -> TVar ChaChaDRG -> NewIntroduction -> m IntroId
+  createIntro SQLiteStore {dbConn} gVar NewIntroduction {toConn, reConn, reInfo} =
+    liftIOEither . createWithRandomId gVar $ \introId ->
+      DB.execute
+        dbConn
+        [sql|
+          INSERT INTO conn_intros
+          (intro_id, to_conn, re_conn, re_info) VALUES (?, ?, ?, ?);
```

Comment:
1 or 2 is okay, but this will quickly spiral out of control. please switch to named parameters, they're just easier to verify than positional

Reply:
You even do named below in update with 4 params